src/Microsoft.Data/StringColumn.cs

```diff
@@ -223,11 +234,15 @@ public override BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapInd
             StringColumn clone;
             if (!(mapIndices is null))
             {
-                if (mapIndices.DataType != typeof(long))
-                    throw new ArgumentException(Strings.MismatchedValueType + " PrimitiveColumn<long>", nameof(mapIndices));
+                if (mapIndices.DataType != typeof(long) && mapIndices.DataType != typeof(bool))
+                    throw new ArgumentException(String.Format("{0} {1} {2} {3}", Strings.MismatchedValueType, "${typeof(long)}", Strings.Or, "${typeof(bool)}", nameof(mapIndices)));
```

Comment:
In general, writing localizable strings this way is not preferred because it doesn't give the localizer a chance to change any word ordering.

It's better to put a full format string in the .resx and just have `{0}` placeholders for specific arguments.

So in this case, the string:

```
"Expected value to be of type {0} or {1}."
```

Should be in the .resx.